great_expectations/data_context/data_context.py

```diff
@@ -953,12 +981,11 @@ def profile_datasource(self, datasource_name, generator_name=None, profiler=Basi
 
                 logger.info("\tProfiled %d rows from %s (%.3f sec)" % (row_count, name, duration))
 
-            #!!! FIXME: THIS IS WAAAAY TO GENERAL. As soon as BatchKwargsError is fully implemented, we'll want to switch to that.
-            # TODO: ^^^
             except ProfilerError as err:
                 logger.warning(err.message)
-            except Exception as exc:
-                logger.warning("\tSomething went wrong when profiling %s. (Perhaps a loading error?) Skipping." % (name))
+            except IOError as exc:
+                logger.warning("\IOError while profiling %s. (Perhaps a loading error?) Skipping." % (name))
```

Comment:
Are we letting all other errors propagate all the way up?

Reply:
I'm not sure yet what other errors might be...but before the behavior was to eat *any* error, which I think of as generally bad practice so I figured we'd let them go up if such errors exist and develop a testing plan once we know what other errors might be